packages/connect/src/device/Device.ts

```diff
@@ -774,9 +772,9 @@ export class Device extends TypedEmitter<DeviceEvents> {
         // TODO: cleanup everything
         _log.debug('Disconnect cleanup');
 
-        this.transportSession = null; // set to null to prevent transport.release
-        this.interruptionFromUser(ERRORS.TypedError('Device_Disconnected'));
-        delete this.runPromise;
+        this.transportSession = null; // set to null to prevent transport.release and cancelableAction
+
+        return this.interruptionFromUser(ERRORS.TypedError('Device_Disconnected'));
```

Comment:
this was the reason of [failing tests](https://github.com/trezor/trezor-suite/pull/13925#issuecomment-2373109255)

`interruptionFromUser` was not awaited and in the same time `runPromise` was deleted therefore if couldn't be rejected by `interruptionFromUser`